src/extension/AutoConversionPrompt.re

```diff
@@ -0,0 +1,362 @@
+          typeTable.remove();
+          el
+        })
+      }
+      | None => {
+        els: [el],
+        text: Element.innerText el,
+        replace: (fun html => { Element.setInnerHTML el html; el })
+      }
+    }
+  });
+
+let getDefTargets =>
+  NotArray.toArray (Document.getElementsByClassName "def")
+  |> Array.map (fun el => {
+    els: [el],
+    text: Element.innerText el,
+    replace: (fun html => {
+      Element.setInnerHTML el ("<pre>" ^ html ^ "</pre>");
+      el
+    })
+  });
+
+let getLstListingTargets =>
+  NotArray.toArray (Document.getElementsByClassName "lstlisting")
+  |> Array.map (fun el => {
+    els: [el],
+    text: Element.innerText el,
+    replace: (fun html => {
+      let parent = Node.parentNode el;
+      Element.setInnerHTML parent ("<pre>" ^ html ^ "</pre>");
+      parent
+    })
+  });
+
+let getCodeTargets =>
+  NotArray.toArray (Document.getElementsByTagName "code")
+  |> Array.map (fun el => {
+    els: [el],
+    text: Element.innerText el,
+    replace: (fun html => {
+      Element.setInnerHTML el html;
+      el
+    })
+  });
+
+let getTargets =>
+  [
+    getPreTargets,
+    getDefTargets,
+    getLstListingTargets,
+    getCodeTargets
+  ]
+  |> List.map (fun f => f ())
+  |> List.map Array.to_list
+  |> List.flatten;
+
+let replaceHrefs hrefs out =>
+  hrefs
+  |> List.fold_left (fun out (text, href) =>
+    Js.String.replaceByRe
+      (Re.make ("/\\b" ^ (escapeRegExp text) ^ "\\b/g"))
+      ("<a href='" ^ href ^"'>" ^ text ^ "</span>")
+      out
+    ) out;
+
+let replaceIds ids out =>
+  ids
+  |> List.fold_left (fun out (text, id) =>
+    Js.String.replaceByRe
+      (Re.make ("/\\b" ^ (escapeRegExp text) ^ "\\b/g"))
+      ("<span class='reason_tools_anchor' id='" ^ id ^"'>" ^ text ^ "</span>")
+      out
+    ) out;
+
+let readjustViewport =>
+  if ((String.length Location.hash) > 0) {
+    Location.reload (Js.Boolean.to_js_boolean true);
+  };
+
+let updateSwapButton => {
+  let style = Element.style syntaxSwap;
+  let reasonLogoUrl = "url(" ^ (Chrome.Extension.getURL reasonLogo) ^ ")";
+  let ocamlLogoUrl = "url(" ^(Chrome.Extension.getURL ocamlLogo) ^ ")";
+
+  Style.setBackgroundImage style
+    (Js.to_bool (Str.includes (Style.backgroundImage style) reasonLogo) ? ocamlLogoUrl : reasonLogoUrl);
+};
+
+type state = { mutable remaining: int };
+
+let swapSyntax type_ => {
+  let targets = getTargets ();
+  let state : state = {
+    remaining: List.length targets
+  };
+
+  ignore (targets
+  |> List.map (fun { els, text, replace } => {
+    open Background;
+    let hrefs = getNormalizedLinks els;
+    let ids = getNormalizedIds els;
+
+    Chrome.Runtime.sendMessage
+      { in_: normalizeText text }
+      (fun maybeResponse => {
+        /* response might, for unknown reasons, sometimes be undefined */
+        switch (Js.Undefined.to_opt maybeResponse) {
+          | None => ()
+          | Some [|[| "Failure", _ |]|] => ()
+          | Some [|[| _, result |]|] => {
```

Comment:
BS intentionally compiles tuples to arrays so you can pattern match on heterogeneous JS arrays with them. I think it'd be a little clearer here. If Refmt gets converted over too we can actually type it.